### PR TITLE
Address escaping in longtext/json

### PR DESCRIPTION
**Pull request by VinaiRachakonda**
Longext is no longer being exported properly in json and csv. This pr fixes that. Cc linked comment.

### CODE REVIEW

**Review by bheni — APPROVED** [review]
lgtm

File: go/libraries/doltcore/table/untyped/csv/writer.go [hunk 1]
```diff
@@ -240,13 +247,13 @@ func (csvw *CSVWriter) write(record []*string) error {
...
 //
-func (csvw *CSVWriter) fieldNeedsQuotes(field *string) bool {
+func fieldNeedsQuotes(field *string) bool {
 	if field != nil && *field == "" {
 		// special Dolt logic
 		return true
 	}
 
-	if *field == `\.` || strings.Contains(*field, csvw.info.Delim) || strings.ContainsAny(*field, "\"\r\n") {
+	if *field == `\.` || strings.Contains(*field, ",") || strings.ContainsAny(*field, "\"\r\n") {
```
Comment (zachmu) [inline go/libraries/doltcore/table/untyped/csv/writer.go:256]: This seems wrong. There are other delimiters possible.

Comment (zachmu) [reply go/libraries/doltcore/table/untyped/csv/writer.go:256]: Should have a test

Comment (VinaiRachakonda) [reply go/libraries/doltcore/table/untyped/csv/writer.go]: Adressed

**Pull request merged**